scripts/run_e2e_tests.py

```diff
@@ -353,64 +398,79 @@ def run_component_template_e2e_test(ctx: Context, template_dir: str) -> bool:
-    remove_if_exists("frontend/cypress/mochawesome")
-    remove_if_exists("frontend/mochawesome-report")
-    remove_if_exists("frontend/mochawesome.json")
+    remove_if_exists("frontend/test_results/cypress")
 
     ctx = Context()
     ctx.always_continue = always_continue
     ctx.record_results = record_results
     ctx.update_snapshots = update_snapshots
     ctx.tests_dir_name = "e2e_flaky" if flaky_tests else "e2e"
 
-    try:
+    def should_run_pretests():
+        # If we're on CircleCI, we intentionally tell CircleCI to send
+        # test files to N-1 of our containers. The Nth container that
+        # doesn't receive anything should run the pretests.
+        if IS_CIRCLECI:
+            return not tests
+
+        # Don't run pretests if we're running flaky tests.
+        return (not flaky_tests) and (not tests)
+
+    def run_pretests():
         # First, test "streamlit hello" in different combinations. We skip
         # `no_credentials=True` for the `--server.headless=false` test, because
         # it'll give a credentials prompt.
-        if (not flaky_tests) and (not test):
-            hello_spec = join(ROOT_DIR, "e2e/specs/st_hello.spec.js")
-            run_test(
-                ctx,
-                hello_spec,
-                ["streamlit", "hello", "--server.headless=true"],
-                no_credentials=False,
-            )
-            run_test(ctx, hello_spec, ["streamlit", "hello", "--server.headless=false"])
-            run_test(ctx, hello_spec, ["streamlit", "hello", "--server.headless=true"])
+        hello_spec = join(ROOT_DIR, "e2e/specs/st_hello.spec.js")
+        run_test(
+            ctx,
+            hello_spec,
+            ["streamlit", "hello", "--server.headless=true"],
+            no_credentials=False,
+        )
+        run_test(ctx, hello_spec, ["streamlit", "hello", "--server.headless=false"])
+        run_test(ctx, hello_spec, ["streamlit", "hello", "--server.headless=true"])
 
-            # Next, run our component_template tests.
-            for template_dir in COMPONENT_TEMPLATE_DIRS:
-                run_component_template_e2e_test(ctx, template_dir)
+        # Next, run our component_template tests.
+        for template_dir in COMPONENT_TEMPLATE_DIRS:
+            run_component_template_e2e_test(ctx, template_dir)
 
+    def run_main_tests():
         # Test core streamlit elements
-        p = pathlib.Path(join(ROOT_DIR, ctx.tests_dir_name, "scripts")).resolve()
-        paths = [p / f"{test}.py"] if test else sorted(p.glob("*.py"))
-        for test_path in paths:
-            test_name, _ = splitext(basename(test_path.as_posix()))
-            specpath = join(ctx.tests_dir, "specs", f"{test_name}.spec.js")
-            run_test(ctx, specpath, ["streamlit", "run", test_path.as_posix()])
+        p = Path(join(ROOT_DIR, ctx.tests_dir_name, "specs")).resolve()
+        paths = [Path(t).resolve() for t in tests] if tests else p.glob("*.spec.js")
```

Comment:
Super minor, but can we sort the `p.glob("*.spec.js")` output (i.e. `sorted(p.glob("*.spec.js"))`), so that running locally always runs the tests in the same order?